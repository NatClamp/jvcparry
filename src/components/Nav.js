import React from 'react';
import { Div, Button } from 'atomize';
import { Link } from 'react-router-dom';

const Nav = () => (
  <>
    <Div d={{ xs: 'none', md: 'flex' }} flexDir="row" justify="space-around" align="center" w={{ xs: 'auto', md: '300px' }} data-testid="nav-container">
      <Link to="/products" style={{ textDecoration: 'none' }}>
        <Button
          h="2.5rem"
          p={{ x: '1rem' }}
          textSize="body"
          textColor="white"
          bg="black"
          border="1px solid"
          borderColor="white"
          hoverTextColor="gray500"
          hoverBorderColor="gray500"
          m={{ r: '0.5rem' }}
        >
          Products
        </Button>
      </Link>
      <Link to="/hire-me" style={{ textDecoration: 'none' }}>
        <Button
          h="2.5rem"
          p={{ x: '1rem' }}
          textSize="body"
          textColor="white"
          bg="black"
          border="1px solid"
          borderColor="white"
          hoverTextColor="gray500"
          hoverBorderColor="gray500"
          m={{ r: '0.5rem' }}
        >
          Hire Me
        </Button>
      </Link>
      <Link to="/blog" style={{ textDecoration: 'none' }}>
        <Button
          h="2.5rem"
          p={{ x: '1rem' }}
          textSize="body"
          textColor="white"
          bg="black"
          border="1px solid"
          borderColor="white"
          hoverTextColor="gray500"
          hoverBorderColor="gray500"
          m={{ r: '0.5rem' }}
        >
          Blog
        </Button>
      </Link>
    </Div>
  </>
);

export default Nav;
