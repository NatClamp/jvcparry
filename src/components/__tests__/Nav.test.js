/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import Nav from '../Nav';

describe('Nav component', () => {
  afterEach(cleanup);
  const history = createMemoryHistory();
  it('should render the Nav component', () => {
    render(
      <Router history={history}>
        <Nav />
      </Router>,
    );
    expect(
      screen.getByTestId('nav-container'),
    ).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <Nav />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
