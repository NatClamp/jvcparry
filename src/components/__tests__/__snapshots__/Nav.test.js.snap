// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component should match the snapshot 1`] = `
<div
  className=""
  data-testid="nav-container"
>
  <a
    href="/products"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className=""
    >
      Products
    </button>
  </a>
  <a
    href="/hire-me"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className=""
    >
      Hire Me
    </button>
  </a>
  <a
    href="/blog"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className=""
    >
      Blog
    </button>
  </a>
</div>
`;
